server/src/main/java/org/elasticsearch/action/support/broadcast/node/TransportBoundedDiagnosticAction.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.action.support.broadcast.node;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.support.ActionFilters;
+import org.elasticsearch.action.support.broadcast.BroadcastRequest;
+import org.elasticsearch.action.support.broadcast.BroadcastResponse;
+import org.elasticsearch.cluster.metadata.IndexNameExpressionResolver;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.util.concurrent.EsRejectedExecutionException;
+import org.elasticsearch.common.util.concurrent.RunOnce;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.transport.TransportService;
+
+/**
+ * A bounded diagnostic action represents an action that aggregates information from all nodes, mainly for diagnostic purposes, and there is
+ * a limit on how many such requests a node can coordinate concurrently. The need for such a class of actions is to protect the node from
+ * getting overwhelmed when another load is slow to respond. The focus is on diagnostic requests because they tend to be executed
+ * periodically.
+ *
+ * @param <Request>              the underlying client request
+ * @param <Response>             the response to the client request
+ * @param <ShardOperationResult> per-shard operation results
+ */
+public abstract class TransportBoundedDiagnosticAction<
```

Comment:
Naming suggestion:

```suggestion
public abstract class TransportBroadcastByNodeStatsAction<
```

The fact that it's bounded is something of an implementation detail, we may want to apply other behaviour across all such stats actions in future too (e.g. tracking how many of them are completed over time)

Reply:
I changed the structure of the code to expand to more stats requests, I chose to create a wrapper that the stats actions could use, in this set-up this naming isn't applicable.